src/index.js

```diff
@@ -0,0 +1,201 @@
+const path				= require('path');
+const log				= require('@whi/stdlog')(path.basename( __filename ), {
+    level: (!__dirname.includes("/node_modules/") && process.env.LOG_LEVEL ) || 'fatal',
+});
+
+// const why				= require('why-is-node-running');
+const net				= require('net');
+const fs				= require('fs');
+const stream				= require('stream');
+const EventEmitter			= require('events');
+
+const { ConversionError,
+	...types }			= require('./types.js');
+const { ...structs }			= require('./structs.js');
+const { MessageParser }			= require('./parser.js');
+const { LairClientError }		= require('./constants.js');;
+
+class TimeoutError extends LairClientError {
+    [Symbol.toStringTag]		= TimeoutError.name;
+}
+
+
+class IncomingRequest {
+
+    constructor ( header, struct, client ) {
+	this._header			= header;
+	this._struct			= struct;
+	this.client			= client;
+    }
+
+    get id () {
+	return this._header.id;
+    }
+
+    get length () {
+	return this._header.length;
+    }
+
+    get wireType () {
+	return this._header.wire_type;
+    }
+
+    get wireTypeId () {
+	return this._header.wire_type_id;
+    }
+
+    get wireTypeStruct () {
+	return this._header.wire_type_class;
+    }
+
+    reply ( ...args ) {
+	let Response			= structs[this.wireTypeId + 1];
+	let wt_resp			= new Response( ...args );
+	let buf				= wt_resp.toMessage( this.id );
+	log.debug("Sending %s response (%s bytes) with ID (%s)", wt_resp.constructor.name, buf.length, this.id );
+	return this.client.send( buf );
+    }
+
+}
+
+class LairClient extends EventEmitter {
+
+    constructor ( address, options ) {
+	super( options );
+
+	const conn			= net.createConnection( address );
+	conn.on('error', function(data) {
+	    console.error(data);
+	    process.exit(1);
+	});
+
+	let connected;
+	this._connect_promise		= new Promise((f,r) => {
+	    connected			= f;
+	});
+	conn.on('connect', () => {
+	    log.silly("Emitting client 'connect' event because of socket 'connect' event");
```

Comment:
Not familiar with this logging framework - is `silly` appropriate? Is that something like trace?

Reply:
`@whi/stdlog` is a wrapper around the [`winston`](https://www.npmjs.com/package/winston) library.  `silly` seems to be a Node ecosystem convention.  My personal preference is `trace` but I wanted to stick to the domain conventions.